altair/vegalite/v5/api.py

```diff
@@ -3865,101 +3850,81 @@ def interactive(  # noqa: C901
+                legend_type = facet_legend.to_dict(context={"data": chart.data})["type"]
+
+            if legend_type == "nominal":
+                # TODO Ideally this would work for ordinal data too
+                legend_selection = selection_point(bind="legend", encodings=[legend])
+                initial_computed_domain = param(expr=f"domain('{legend}')")
+                nonreactive_domain = param(
+                    react=False, expr=initial_computed_domain.name
                 )
-
-            if legend_encoding is not None:
-                if utils.is_undefined(
-                    interactive_chart.encoding[legend_encoding]["type"]
-                ):
-                    legend_encoding_type = interactive_chart.encoding[
-                        legend_encoding
-                    ].to_dict(context={"data": interactive_chart.data})["type"]
+                scale = facet_legend["scale"]
+                if utils.is_undefined(scale):
+                    scale = {"domain": nonreactive_domain}
                 else:
-                    legend_encoding_type = interactive_chart.encoding[legend_encoding][
-                        "type"
-                    ]
-                if (
-                    legend_encoding_type == "nominal"
-                ):  # TODO Ideally this would work for ordinal data too
-                    legend_selection = selection_point(
-                        bind="legend", encodings=[legend_encoding]
-                    )
-                    initial_computed_domain = param(expr=f"domain('{legend_encoding}')")
-                    nonreactive_domain = param(
-                        react=False, expr=initial_computed_domain.name
-                    )
-                    if utils.is_undefined(
-                        interactive_chart.encoding[legend_encoding]["scale"]
-                    ):
-                        interactive_chart.encoding[legend_encoding]["scale"] = {
-                            "domain": nonreactive_domain
-                        }
-                    else:
-                        interactive_chart.encoding[legend_encoding]["scale"][
-                            "domain"
-                        ] = nonreactive_domain
-
-                    interactive_chart = interactive_chart.add_params(
-                        legend_selection, initial_computed_domain, nonreactive_domain
-                    ).transform_filter(legend_selection)
-        return interactive_chart
+                    scale["domain"] = nonreactive_domain
+                chart = chart.add_params(
+                    legend_selection,
+                    initial_computed_domain,
+                    nonreactive_domain,
+                ).transform_filter(legend_selection)
+            else:
+                msg = f"Expected only 'nominal' legend type but got {legend_type!r}"
+                raise NotImplementedError(msg)
+        return chart
+
+
+LegendChannel_T: TypeAlias = Literal[
```

Comment:
This is the singular definintion you can reuse.

If any of the *TODO*s get resolved, everything will stay in sync w/ this.

Reply:
> but didn't know of `typing.get_args()` , that's very helpful!

It is quite handy for situations like this, where during development it isn't known what the final set will be.

Since it is new to you, I'd say try not rely on it too much.
You might run into issues using it on types declared **outside of the current module** - see https://github.com/vega/altair/pull/3516#discussion_r1703151552 - which a lot of `Literal`(s) are.

I would suggest that once the final `LegendChannel_T` definition is decided, then add a either a `frozenset` (if the order *is not* important) or a `tuple` (if the order *is* important) containing those elements.
It would also be faster, but not sure that is much of a concern here